src/mechanics/ingame/demonslayer/giyu/TomiokaGiyu.java

```diff
@@ -1,37 +1,82 @@
 package mechanics.ingame.demonslayer.giyu;
 
+import mechanics.Texts;
 import mechanics.abilities.Ability;
 import mechanics.characters.Character;
 
 public class TomiokaGiyu extends Character {
+	
+	private int instantKillBoostFromAbility1;
+	private int instantKillBoostFromAbility2;
 
 	public TomiokaGiyu(int id) {
 		super(id);
-		// TODO Auto-generated constructor stub
+		
+		this.instantKillBoostFromAbility1 = 0;
+		this.instantKillBoostFromAbility2 = 0;
+	}
+	
+	private void clearInstantKillBuffs() {
+		this.removeActive(this.getA1());
+		this.removeActive(this.getA2());
+		
+		this.instantKillBoostFromAbility1 = 0;
+		this.instantKillBoostFromAbility2 = 0;
 	}
 
 	@Override
-	public void applyAbility1(Ability a, Character c) {
-		// TODO Auto-generated method stub
+	public void applyAbility1(Ability ability, Character target) {
+		if (this.instantKillBoostFromAbility1 == 0) {
+			String[] abilityText = {"Water Breathing 11th Form: Dead Calm is improved"};
```

Comment:
When you apply the 1st and 2nd skill, it should appear that the 3rd skill is improved, and not the 1st nor 2nd Ability. Theres already a method in Texts that does this:
```suggestion
			String[] abilityText = {Texts.SKILL_IMPROVED(this.getA3().getName())};
```